dttools/src/stringtools.c

```diff
@@ -46,6 +47,27 @@ char *escape_shell_string(const char *str)
 	return escaped_string;
 }
 
+char *string_escape_shell( const char *str )
+{
+	buffer_t buffer;
```

Comment:
I find it helpful to use an array here so you don't need to use `&` everywhere:

`buffer_t B[1];`

then you can do:

`buffer_putliteral(B, "foo");`